agent/proxycfg/testing.go

```diff
@@ -1828,7 +1883,7 @@ func testConfigSnapshotTerminatingGateway(t testing.T, populateServices bool) *C
 
 		snap.TerminatingGateway.ServiceConfigs = map[structs.ServiceName]*structs.ServiceConfigResponse{
 			web: {
-				ProxyConfig: map[string]interface{}{"protocol": "tcp"},
+				ProxyConfig: map[string]interface{}{"protocol": "http"},
```

Comment:
Is the intent here that a single `testing.go` helper exists for manufacturing terminating gateway setups, and because it points to both web (http) and api (tcp) the generated testdata will showcase both at the same time?

Reply:
Right, though I just moved the http override into the relevant test cases. I think I like having explicit overrides in the test case instead.